src/main/scala/org/neo4j/spark/reader/Neo4jDataSourceReader.scala

```diff
@@ -19,20 +21,43 @@ class Neo4jDataSourceReader(private val options: DataSourceOptions, private val
   private val neo4jOptions: Neo4jOptions = new Neo4jOptions(options.asMap())
     .validate(options => Validations.read(options, jobId))
 
-  override def readSchema(): StructType = {
-    val schemaService = new SchemaService(neo4jOptions, jobId)
+
+  override def readSchema(): StructType = callSchemaService { schemaService => schemaService
+    .struct() }
+
+  private def callSchemaService[T](lambda: SchemaService => T): T = {
+    val driverCache = new DriverCache(neo4jOptions.connection, jobId)
+    val schemaService = new SchemaService(neo4jOptions, driverCache)
+    var hasError = false
     try {
-      schemaService.struct()
+      lambda(schemaService)
+    } catch {
+      case e: Throwable => {
+        hasError = true
```

Comment:
logging?

Reply:
It will logged by Spark, I can add it but it would be redundant